### PR TITLE
Updated from Jade to Pug

### PR DESCRIPTION
**Pull request by benjamin-richardson**
I have updated the Jade parts of Yeogurt to now reference Pug. Soon the Jade name will die off completely and it is no longer be developed under that name. This will make the generator easier to update and use moving into the future.

Slight backwards compatibility issue with the sub-generators. Not sure if this is really an issue or just perceived issues.

### CODE REVIEW

File: app/templates/gulp/es5/pug.js [hunk 1]
```diff
@@ -49,13 +49,13 @@ module.exports = function(gulp, plugins, args, config, taskTarget, browserSync)
...
-      path.join(dirs.source, '**/*.jade'),
+      path.join(dirs.source, '**/*.pug'),
       '!' + path.join(dirs.source, '{**/\_*,**/\_*/**}')
     ])
     .pipe(plugins.changed(dest))
     .pipe(plugins.plumber())
-    .pipe(plugins.jade({
-      jade: jade,
+    .pipe(plugins.pug({
+      pug: pug,
```
Comment (Bitneko) [inline app/templates/gulp/es5/pug.js]: Gulp-pug don't require this option. Gulp task fail when this option is defined

File: app/templates/gulp/es6/pug.js [hunk 2]
```diff
@@ -49,13 +49,13 @@ export default function(gulp, plugins, args, config, taskTarget, browserSync) {
...
-      path.join(dirs.source, '**/*.jade'),
+      path.join(dirs.source, '**/*.pug'),
       '!' + path.join(dirs.source, '{**/\_*,**/\_*/**}')
     ])
     .pipe(plugins.changed(dest))
     .pipe(plugins.plumber())
-    .pipe(plugins.jade({
-      jade: jade,
+    .pipe(plugins.pug({
+      pug: pug,
```
Comment (Bitneko) [inline app/templates/gulp/es6/pug.js]: gulp-pug don't require this option. Gulp task fail when this option is defined

**Comment by larsonjj** [comment]
Hey @benjamin-richardson,

Thanks for the PR! Apologies for the late response.

If you could fix the conflicts with new release, that would be awesome. As for merging this in, I'm waiting for pug to come out of beta. Until it's full release, I will keep this PR open and unmerged.

Let me know if you have any questions. Nice work 👍

File: app/templates/gulp/es5/pug.js (hunk 1 above)
Comment (larsonjj) [inline app/templates/gulp/es5/pug.js]: Looks like the `pug` option is available ([source](https://www.npmjs.com/package/gulp-pug#pugopts)). So feel free to ignore making any changes @benjamin-richardson

File: app/templates/gulp/es6/pug.js (hunk 2 above)
Comment (larsonjj) [inline app/templates/gulp/es6/pug.js]: Looks like the `pug` option is available ([source](https://www.npmjs.com/package/gulp-pug#pugopts)). So feel free to ignore making any changes @benjamin-richardson

**Comment by Dan503** [comment]
I think this issue on the Pug GitHub Repo says that it has officially been released now:
https://github.com/pugjs/pug/issues/2776

Does that mean that this Pull request can be merged now?

**Comment by larsonjj** [comment]
@Dan503, Looks like it's still in rc-4, so I'm still going to hold off on updating from Jade for a bit longer. Not sure why it's taking so long for a stable release, as it is quite frustrating for this project. Hopefully a stable release will be pushed soon and I will get it integrated with Yeogurt quickly.

**Comment by stasguma** [comment]
Maybe it`s time? Already 2.0.3 version.
https://github.com/pugjs/pug/releases

**Comment by larsonjj** [comment]
Closing this in favor of #226 as the code required some pretty substantial surgery to get things up to date and working.

**Pull request closed without merging**